clearblade/cloud/iot_v1/devices.py

```diff
@@ -0,0 +1,619 @@
+
+    @property
+    def device_ids(self):
+        return self._device_ids
+
+    @property
+    def field_mask(self):
+        return self._field_mask
+
+    @property
+    def gateway_list_options(self):
+        return self._gateway_list_options
+
+    @property
+    def page_size(self):
+        return self._page_size
+
+    @property
+    def page_token(self):
+        return self._page_token
+
+    def _prepare_params_for_list(self):
+        params = {'parent':self.parent}
+        if self.page_size is not None:
+            params['pageSize'] = self.page_size
+        if self.device_num_ids is not None:
+            params['deviceNumIds'] = self.device_num_ids
+        if self.device_ids is not None:
+            params['deviceIds'] = self.device_ids
+        if self.field_mask is not None:
+            params['fieldMask'] = self.field_mask
+        if self.gateway_list_options is not None:
+            params['gatewayListOptions'] = self.gateway_list_options
+        if self.page_token is not None:
+            params['pageToken'] = self.page_token
+
+class UpdateDeviceRequest(Request):
+    def __init__(self, id: str = None, name: str = None, numId: str=None,
+                 credentials: list=None, blocked: bool = None,
+                 logLevel: str = None, metadata: dict = None, gatewayConfig : dict = None,
+                 updateMask: str = None) -> None:
+        self._id = id
+        self._name = name
+        self._num_id = numId
+        self._credentials = credentials
+        self._blocked = blocked
+        self._log_level = logLevel
+        self._meta_data = metadata
+        self._gateway_config = gatewayConfig
+        self._update_mask = updateMask
+
+    def _prepare_params_body_for_update(self):
+        params = {'name': self._name}
+        if self._update_mask is not None:
+            params['updateMask'] = self._update_mask
+
+        body = {}
+        if self._id is not None:
+            body['id'] = self._id
+        if self._name is not None:
+            body['name'] = self._name
+        if self._log_level is not None:
+            body['logLevel'] = self._log_level
+        if self._gateway_config is not None:
+            body['gatewayConfig'] = self._gateway_config
+        if self._meta_data is not None:
+            body['metadata'] = self._meta_data
+        if self._blocked is not None:
+            body['blocked'] = self._blocked
+        if self._credentials is not None:
+            body['credentials'] = self._credentials
+
+        return params, body
+
+class ListDevicesResponse():
+
+    @property
+    def raw_pages(self):
+        pass
+
+    def devices(self):
+        pass
+
+    def next_page_token(self):
+        pass
+
+class ClearBladeDeviceManager():
+
+    def __init__(self) -> None:
+        #create the ClearBladeConfig object
+        self._config_manager = ClearBladeConfigManager()
+        self._config_manager.registry_name = "gargi_python"
```

Comment:
The test samples are failing due to hardcoded registry_name.

Reply:
@rajasd27  i have updated the PR , and now you need to set CLEARBLADE_REGISTRY as an environment variable and mention the registry name there. 

Also just to remind you, you also need to CLEARBLADE_CONFIGURATION as an environment variable as well which points to the service config json file.

Reply:
Perfect, thanks @gargi-ingresssoftware. Also, we'll need to add another env variable for the registry location. Currently, the registry location is also hardcoded [here](https://github.com/ClearBlade/python-iot/pull/6/files#diff-6d27665e7aa62323db233773a1205975471b715d01cd4e0724d4bc408caf2af8R11)  and it fails when trying to fetch registry information from other regions (asia-east1 & europe-west1).

Reply:
Done